### PR TITLE
Test everything from root workspace

### PR DESCRIPTION
**Pull request by gregmagolan**
* Move nested workspaces from /internal/e2e to /e2e and test with bazel_integration_test
* Test examples with bazel_integration_test
* Remove lots of old shell scripts that are no longer needed since everything is testable from root workspace and no more release artifacts needed to test nested WORKSPACES as this is handled by bazel_integration_test
* Lots of cleanup related to above

### CODE REVIEW

File: e2e/bazel_managed_deps/deps.spec.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ describe('dependencies', () => {
     const files = ['jasmine', 'tsc', 'tsserver'];
     for (const f of files) {
       try {
-        expect(require.resolve(`.bin/${f}`).endsWith(`/npm/node_modules/.bin/${f}`)).toBe(true);
+        expect(!!require.resolve(`.bin/${f}`)).toBe(true);
```
Comment (alexeagle) [inline e2e/bazel_managed_deps/deps.spec.js]: nit: could use `toBeTruthy`

File: e2e/user_managed_deps/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# User managed dependencies
+
+This example shows how to write a simple Node.js program using nodejs rules from source and user managed npm dependencies.
```
Comment (alexeagle) [inline e2e/user_managed_deps/README.md]: maybe this one should be under examples/ ?

File: scripts/clean_all.sh [hunk 3]
```diff
@@ -10,25 +10,20 @@ cd ${RULES_NODEJS_DIR}
...
+readonly workspaceRoots=("e2e" "examples" "packages")
+for workspaceRoot in ${workspaceRoots[@]} ; do
+  (
+    readonly workspaceFiles=($(find ./${workspaceRoot} -type f -name WORKSPACE -prune))
+    for workspaceFile in ${workspaceFiles[@]} ; do
+      (
+        readonly workspaceDir=$(dirname ${workspaceFile})
+        printf "\n\nCleaning ${workspaceDir}\n"
+        cd ${workspaceDir}
         echo_and_run bazel clean --expunge
```
Comment (alexeagle) [inline scripts/clean_all.sh]: do you think we actually need this?

File: scripts/clean_yarn_cache_selectively.sh [hunk 4]
```diff
@@ -7,18 +7,18 @@ set -eu -o pipefail
...
-  echo_and_run yarn cache clean @bazel/${package}
-done
+readonly NPM_PACKAGE_LABELS=`bazel query --output=label 'kind("npm_package", //packages/...)'`
 
-# Also clean cache for different versions of yarn since the locally installed
-# yarn may have a different cache version from yarn version used by Bazel
-for package in ${PACKAGES[@]} ; do
+for npmPackageLabel in ${NPM_PACKAGE_LABELS[@]} ; do
+  # Trim //packages/foobar:npm_package to foobar
+  package=$(echo ${npmPackageLabel} | rev | cut -c 13- | rev | cut -c 12-)
```
Comment (alexeagle) [inline scripts/clean_yarn_cache_selectively.sh]:
```
$ bazel query --output=package 'kind("npm_package", //packages/...)'
packages/create
packages/hide-bazel-files
packages/jasmine
packages/karma
packages/labs
packages/protractor
packages/stylus
packages/typescript
```

File: scripts/test_all.sh [hunk 5]
```diff
@@ -28,11 +28,11 @@ echo_and_run bazel run @nodejs//:yarn
 printf "\n\nBuilding all targets\n"
 echo_and_run bazel build ...
 
-printf "\n\nTesting all targets\n"
+printf "\n\nTesting all targets (except integration tests)\n"
```
Comment (alexeagle) [inline scripts/test_all.sh]: why do we still have this shell script?

File: scripts/clean_yarn_cache_selectively.sh (hunk 4 above)
Comment (gregmagolan) [reply scripts/clean_yarn_cache_selectively.sh]: 👍 thanks

File: scripts/test_all.sh (hunk 5 above)
Comment (gregmagolan) [reply scripts/test_all.sh]: It's useful to make sure that all assertions in CI are easy to run locally. For most purposes the following will do locally:
```
yarn test
yarn test_e2e
yarn test_examples
```

File: scripts/clean_all.sh (hunk 3 above)
Comment (gregmagolan) [reply scripts/clean_all.sh]: Changed to clean_nested_workspaces.sh. Useful if you cd into a nested workspaces and run a it manually and want to clean it up after from root. If there is a `dist` folder left over in the nested workspace it messes up the root workspace's build & test.

File: e2e/bazel_managed_deps/deps.spec.js (hunk 1 above)
Comment (gregmagolan) [reply e2e/bazel_managed_deps/deps.spec.js]: 👍

File: e2e/user_managed_deps/README.md (hunk 2 above)
Comment (gregmagolan) [reply e2e/user_managed_deps/README.md]: 👍 moved. plus e2e is getting slow on CI so that's better. should shard the e2e tests in CI in follow up PR.

**Pull request merged**